node/__tests__/address.test.ts

```diff
@@ -0,0 +1,83 @@
+import {
+  PlaceDetailsRequest,
+  PlaceDetailsResponse,
+  Status,
+} from '@googlemaps/google-maps-services-js'
+import { Apps, IOClients, IOContext } from '@vtex/api'
+import { QueryAddressArgs } from 'vtex.geolocation-graphql-interface'
+
+import { Clients } from '../clients'
+import { Google } from '../clients/Google'
+import getAddress from '../resolvers/address'
+
+const mockArgs = (): QueryAddressArgs => ({
+  externalId: 'id',
+  sessionToken: 'session token',
+})
+
+interface MockContextProps {
+  locale: string
+  warn: (warning: unknown) => void
+}
+
+const mockContext = ({ locale, warn }: MockContextProps): Context =>
+  ({
+    clients: {
+      google: {
+        placeDetails: (_request: PlaceDetailsRequest) =>
+          Promise.resolve({
+            statusText: Status.OK,
+            data: {},
+          } as PlaceDetailsResponse),
+      } as Google,
+      apps: {
+        getAppSettings: (_app: string) =>
+          Promise.resolve({
+            apiKey: 'api key',
+          } as { apiKey: string }),
+      } as Apps,
+    } as IOClients & Clients,
+    vtex: {
+      locale,
+      logger: {
+        warn,
+        error: (_error: unknown) => undefined,
+      },
+    } as IOContext,
+  } as Context)
+
+describe('address', () => {
+  it.each(['pt-BR', 'en'])(
+    'shouldn\'t log "%s" as invalid language code',
+    async (language: string) => {
+      const mockWarn = jest.fn((_warning: unknown) => undefined)
+
+      await getAddress(
+        {},
+        mockArgs(),
+        mockContext({ locale: language, warn: mockWarn })
+      )
+
+      expect(mockWarn).not.toBeCalledWith(
+        `"${language}" is not a valid language. See the list of supported languages on https://developers.google.com/maps/faq#languagesupport`
+      )
```

Comment:
what do you think of adding only `expect(mockWarn).not.toBeCalled()`, so if we change the message we will ensure the test will keep working correctly

Reply:
what if there are other warnings? I suppose we could just revert your other suggestion, right? `expect(mockWarn.mock.calls[0]).not.toMatchInlineSnapshot(...)`

Reply:
I think another approach would be standardize the messages in a different file and do something like that instead:

```typescript
// someUtil.ts
function getInvalidLanguageWarning(language: string) {
  return `"${language}" is not a valid language. See the list of supported languages on https://developers.google.com/maps/faq#languagesupport`
}
```

And use it on both resolver and testing file. Another small advantage here, is that two resolvers actually use this same message.

Reply:
hmm, don't know how I feel about the function. Seems a bit excessive to me :-/

> what if there are other warnings? I suppose we could just revert your other suggestion, right? `expect(mockWarn.mock.calls[0]).not.toMatchInlineSnapshot(...)`

this wouldn't work if we logged something else before this log :-/

we could add one assertion to the number of times this function was called, so whenever we add a new log in this resolver we'd need to update the test accordingly.

Reply:
I see, but I think that this will turn out to be trickier and will end up testing other things besides what is expected.

1. If more warnings get added or removed, this test will break
2. If undesired warnings start to get called/uncalled, this test will break with other tests

Even though I also don't really like the raw string comparison, if it breaks, I think it will be very clear where and why the test broke. What do you think?

Reply:
yeah, I don't know a good solution in this case :-(

Reply:
Yep :^/
I was thinking about adding identifiers to logs, but that also looks kinda bad and verbose.
Can we keep my proposal (the committed code) for now? I'll take a look on how to improve this